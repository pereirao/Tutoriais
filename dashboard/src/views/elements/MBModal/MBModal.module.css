.overlay {
  position: absolute;
  width: 100%;
  height: 4000px;
  top: 0px;
  left: 0px;
  z-index: 9998;
  background: rgba(0,0,0,0.85);
}

.modalWrapper {
  position: absolute;
  width: 340px;
  display: flex;
  flex-direction: column;
  align-items: center;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 9999;
  background: #FFF;
  padding: 16px;
}

.modalWrapper.large {
  width: 480px;
}

@media only screen and (max-width: 480px) and (min-width: 0) {
  .modalWrapper.large {
    width: 320px;
  }
}

.closeWrapper {
  align-self: flex-start;
  height: 16px;
  width: 16px;
  border-radius: 8px;
  cursor: pointer;
}

.closeWrapper:hover {
  background-color: #9B9B9B;
}

.closeIcon {
  position: relative;
  top: 2px;
  left: 2px;
}

.closeWrapper:hover .closeIcon line{
  stroke: #FFF;
}

