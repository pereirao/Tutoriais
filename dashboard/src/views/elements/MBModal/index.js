// @flow

import React, {PureComponent} from 'react';
import classNames from 'classnames/bind';
import style_sheet from './MBModal.module.css';

const cx = classNames.bind(style_sheet);

type MBModalProps = { close: () => void, size: string, closeable: boolean, children: any };

class MBModal extends PureComponent {
  props: MBModalProps
  static defaultProps = {size: 'medium', closeable: true}

  componentWillMount() {
    const body_el = document.getElementById('body');
    if (body_el) body_el.style.overflow = 'hidden';
    // document.ontouchmove = e => { e.preventDefault(); }; // Disable scrolling. FIXME: currently breaks mobile dropdown select
  }

  componentWillUnmount() {
    const body_el = document.getElementById('body');
    if (body_el) body_el.style.overflow = 'auto';
    // document.ontouchmove = _e => { return true; }; // Enable scrolling. FIXME: currently breaks mobile dropdown select
  }

  handleOverlayClick = (e) => {
    e.preventDefault();
    if (this.props.closeable) this.props.close();
  };

  render() {
    const {close, size, closeable, children} = this.props;
    return (
      <div className={style_sheet.container}>
        <div className={cx('modalWrapper', size)}>
          {
            closeable &&
            <span
              onClick={close}
              role="button"
              tabIndex="0"
              className={style_sheet.closeWrapper}>
              <CloseIcon/>
            </span>
          }
          {children}
        </div>
        <div className={style_sheet.overlay} role="presentation" onClick={this.handleOverlayClick}/>
      </div>
    );
  }
}

const CloseIcon = () => (
  <svg className={style_sheet.closeIcon} height="16px" width="16px" version="1.1" xmlns="http://www.w3.org/2000/svg">
    <line x1="1" y1="11" x2="11" y2="1" stroke="#781214" strokeWidth="2"/>
    <line x1="1" y1="1" x2="11" y2="11" stroke="#781214" strokeWidth="2"/>
  </svg>
);

export default MBModal;
