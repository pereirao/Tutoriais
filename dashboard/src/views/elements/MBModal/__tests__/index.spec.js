import React from 'react';

import MBModal from '../index';

it('renders correctly', () => {
  expect(shallow(<MBModal />)).toMatchSnapshot();
});
