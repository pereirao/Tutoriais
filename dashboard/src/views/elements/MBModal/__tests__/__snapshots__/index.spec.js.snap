// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="modalWrapper medium"
  >
    <span
      className="closeWrapper"
      role="button"
      tabIndex="0"
    >
      <CloseIcon />
    </span>
  </div>
  <div
    className="overlay"
    onClick={[Function]}
    role="presentation"
  />
</div>
`;
